sdks/python/apache_beam/testing/util.py

```diff
@@ -154,6 +151,8 @@ def matches_all(expected):
       the elements of a single PCollection.
   """
   def _matches(actual):
+    from hamcrest.core import assert_that as hamcrest_assert
```

Comment:
Would this break users if they use assert and they are not aware that they need to pre-install hamcrest as an extra dependency?

Reply:
yes. This would break them if they use this specific assertion, and did not install beam's test dependencies via `pip install apache-beam[test]`, or install hamcrest directly via `pip install pyhamcrest`

Reply:
The same would happen if users use other Beam utilities such as `bigquery_matcher.py`, or `pipeline_verifiers.py`

Reply:
`pipeline_verifiers.py` is a good argument. (Which btw. does not do dynamic loading)

Should we say that, if you are using testing/* module you need to install [test] extra packages? And also change pipeline_verifiers.py to use dynamic loading as well?